test/compress/loops.js

```diff
@@ -240,3 +240,83 @@ issue_1532: {
         }
     }
 }
+
+issue_186: {
+    beautify = {
+        beautify: false,
+        screw_ie8: true,
+    }
+    input: {
+        var x = 3;
+        if (foo()) {
+            do {
```

Comment:
In all the do-while tests in this PR can you remove the braces from `input:`?

The uglify parser should handle that correctly.